### ISSUE TITLE
Many threads panicked at block 100000

### ISSUE THREAD

**Issue by bladedoyle** [mimblewimble/grin#1148]
Very long log at https://ipfs.io/ipfs/QmPgEFyQgecL6hQaCYX2rYeJWKVeDswwRHqCSATrgifAzx
```

Jun 07 05:35:48.209 ERRO
thread 'compactor' panicked at 'slice index starts at 2516 but ends at 2482': libcore/slice/mod.rs:791stack backtrace:
   0:     0x558245aa15bc - backtrace::backtrace::trace::heaba8652fc04e149
   1:     0x558245a9fd42 - backtrace::capture::Backtrace::new::hfe8602475f4016d0
   2:     0x558245a268af - grin_util::logger::send_panic_to_log::{{closure}}::hd7372461a8c4ee89
   3:     0x558245b5a4d9 - std::panicking::rust_panic_with_hook::hed3a578f3d672948
                        at libstd/panicking.rs:403
   4:     0x558245b5a2c2 - std::panicking::begin_panic_fmt::h3e8fdd00db89fef8
                        at libstd/panicking.rs:349
   5:     0x558245b5a1f2 - rust_begin_unwind
                        at libstd/panicking.rs:325
   6:     0x558245ba9cb0 - core::panicking::panic_fmt::h16b501044b4579f2
                        at libcore/panicking.rs:72
   7:     0x558245baa644 - core::slice::slice_index_order_fail::h70fdd58de97ef27c
                        at libcore/slice/mod.rs:791
   8:     0x558245806fb0 - grin_store::types::AppendOnlyFile::save_prune::hb65856bb6a25e1c4
   9:     0x5582457ffb76 - <grin_store::pmmr::PMMRBackend<T>>::check_compact::h2e0b1081bba07219
  10:     0x55824581a6c8 - grin_chain::txhashset::TxHashSet::compact::h903fa8b43ced0c71
  11:     0x558245812cca - grin_chain::chain::Chain::compact::hf2a2d569a999f6c1
  12:     0x55824552a06a - std::sys_common::backtrace::__rust_begin_short_backtrace::h75786730fb99ad3a
  13:     0x558245b6d98e - __rust_maybe_catch_panic
                        at libpanic_unwind/lib.rs:102
  14:     0x558245554c47 - <F as alloc::boxed::FnBox<A>>::call_box::h5a840c094950260b
  15:     0x558245b62b27 - <alloc::boxed::Box<alloc::boxed::FnBox<A, Output$u3d$R$GT$$u20$$u2b$$u20$$u27$a$GT$$u20$as$u20$core..ops..function..FnOnce$LT$A$GT$$GT$::call_once::h3a46ab40d44ff8db
                        at /checkout/src/liballoc/boxed.rs:794
                         - std::sys_common::thread::start_thread::heca3dbba8ff4f46a
                        at libstd/sys_common/thread.rs:24
  16:     0x558245b4eae8 - std::sys::unix::thread::Thread::new::thread_start::ha1ecaaa6d3efa77d
                        at libstd/sys/unix/thread.rs:90
  17:     0x7f9bb197d7fb - start_thread
  18:     0x7f9bb1493b5e - clone
  19:                0x0 - <unknown>


Jun 07 05:35:48.209 ERRO
thread 'stratum_server' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"': libcore/result.rs:945stack backtrace:
   0:     0x558245aa15bc - backtrace::backtrace::trace::heaba8652fc04e149
   1:     0x558245a9fd42 - backtrace::capture::Backtrace::new::hfe8602475f4016d0
   2:     0x558245a268af - grin_util::logger::send_panic_to_log::{{closure}}::hd7372461a8c4ee89
   3:     0x558245b5a4d9 - std::panicking::rust_panic_with_hook::hed3a578f3d672948
                        at libstd/panicking.rs:403
   4:     0x558245b5a2c2 - std::panicking::begin_panic_fmt::h3e8fdd00db89fef8
                        at libstd/panicking.rs:349
   5:     0x558245b5a1f2 - rust_begin_unwind
                        at libstd/panicking.rs:325
   6:     0x558245ba9cb0 - core::panicking::panic_fmt::h16b501044b4579f2
                        at libcore/panicking.rs:72
   7:     0x5582457e4e3e - core::result::unwrap_failed::h089c689469fad438
   8:     0x55824581106a - grin_chain::chain::Chain::set_txhashset_roots::hb3cc8a8296284d03
   9:     0x558245521957 - grin_servers::mining::mine_block::build_block::h9c50659b3ad24f84
  10:     0x55824551f3b1 - grin_servers::mining::mine_block::get_block::hc30dbb539e588894
  11:     0x5582455655e0 - grin_servers::mining::stratumserver::StratumServer::run_loop::ha56391b8d871ddc3
  12:     0x55824552b705 - std::sys_common::backtrace::__rust_begin_short_backtrace::h90a19b1cfd3e089e
  13:     0x558245566d55 - std::panicking::try::do_call::ha9d39411ab0a49ac
  14:     0x558245b6d98e - __rust_maybe_catch_panic
                        at libpanic_unwind/lib.rs:102
  15:     0x558245554fc0 - <F as alloc::boxed::FnBox<A>>::call_box::h8570018d411845be
  16:     0x558245b62b27 - <alloc::boxed::Box<alloc::boxed::FnBox<A, Output$u3d$R$GT$$u20$$u2b$$u20$$u27$a$GT$$u20$as$u20$core..ops..function..FnOnce$LT$A$GT$$GT$::call_once::h3a46ab40d44ff8db
                        at /checkout/src/liballoc/boxed.rs:794
                         - std::sys_common::thread::start_thread::heca3dbba8ff4f46a
                        at libstd/sys_common/thread.rs:24
  17:     0x558245b4eae8 - std::sys::unix::thread::Thread::new::thread_start::ha1ecaaa6d3efa77d
                        at libstd/sys/unix/thread.rs:90
  18:     0x7f9bb197d7fb - start_thread
  19:     0x7f9bb1493b5e - clone
  20:                0x0 - <unknown>
```

**Comment by ignopeverell** [comment]
Did you have #1143 when this happened?

**Comment by bladedoyle** [comment]
No, was running code cloned Jun 4

**Comment by ignopeverell** [comment]
Ok thanks, feel free to reopen if it occurs again but this was likely fixed in #1143.
